src/functions.ts

```diff
@@ -40,6 +40,19 @@ export type HandlerFunction =
   | CloudEventFunction
   | CloudEventFunctionWithCallback;
 
+/**
+ * A legacy event.
+ */
+export interface LegacyEvent {
+  data: object;
+  context: CloudFunctionsContext;
+}
+
+interface Data {
+  data: object;
+}
+export type LegacyCloudFunctionsContext = CloudFunctionsContext | Data;
```

Comment:
Is the `|` here because there are scenarios where the context argument to a user's function a `Data`?

Reply:
Correct, iirc there were certain cases where the context argument was a data object.